sympy/core/tests/test_sympify.py

```diff
@@ -512,6 +513,51 @@ def test_sympify_set():
     assert sympify(set()) == EmptySet()
 
 
+def test_numpy():
+    from sympy.utilities.pytest import skip
+    np = import_module('numpy')
+
+
+    def equal(x, y):
+        return x == y and type(x) == type(y)
+
+    if not np:
+        skip('numpy not installed.Abort numpy tests.')
+
+    assert sympify(np.bool_(1)) is S(True)
+    assert equal(
+        sympify(np.int_(1234567891234567891)), S(1234567891234567891))
+    assert equal(sympify(np.intc(1234567891)), S(1234567891))
+    assert equal(
+        sympify(np.intp(1234567891234567891)), S(1234567891234567891))
+    assert equal(sympify(np.int8(-123)), S(-123))
+    assert equal(sympify(np.int16(-12345)), S(-12345))
+    assert equal(sympify(np.int32(-1234567891)), S(-1234567891))
+    assert equal(
+        sympify(np.int64(-1234567891234567891)), S(-1234567891234567891))
+    assert equal(sympify(np.uint8(123)), S(123))
+    assert equal(sympify(np.uint16(12345)), S(12345))
+    assert equal(sympify(np.uint32(1234567891)), S(1234567891))
+    assert equal(
+        sympify(np.uint64(1234567891234567891)), S(1234567891234567891))
+    assert equal(sympify(np.float16(1.123)), S(1.123))
+    assert equal(sympify(np.float32(1.1234567)), S(1.1234567))
+    assert equal(sympify(np.float64(1.1234567891234)), S(1.1234567891234))
```

Comment:
I don't think any precision is lost, since the default Float precision corresponds (roughly) to a 64-bit machine float (53 bits, which you can see from the `_prec` attribute of Float). But it would seem to me that perhaps the lower floats should create a float with less precision. For instance, a 32-bit float only has 24 bits of precision (7 decimal digits). See https://en.wikipedia.org/wiki/Floating_point#Internal_representation.

Float takes a decimal precision, which isn't exact. It may be better to convert the float to the appropriate `_mpf_` tuple. Even so, the Float constructor ignores the fourth field in a tuple over the decimal prec. It would be nice to either change that (I think I tried once but got some test failures), or add support to pass in binary precision directly to the Float constructor.